File: onlinepayments-sdk/src/main/java/com/onlinepayments/sdk/client/android/model/paymentproduct/Tooltip.java
/*
 * Copyright 2020 Global Collect Services B.V
 */

package com.onlinepayments.sdk.client.android.model.paymentproduct;

import android.graphics.drawable.Drawable;

import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

/**
 * POJO that represents a Tooltip object.
 * Tooltips are payment product specific and are used to show extra information about an input field.
 */
public class Tooltip implements Serializable {

	private static final long serialVersionUID = -317203058533669043L;

	private String label;
	/**
	 * @deprecated In a future release, this property will be removed since it is not returned from the API.
	 */
	@Deprecated
	@SerializedName("image")
	private String imageURL;
	/**
	 * @deprecated In a future release, this property will be removed since it is not returned from the API.
	 */
	@Deprecated
	private transient Drawable imageDrawable;

	/**
	 * @deprecated In a future release, this constructor will become internal to the SDK.
	 */
	@Deprecated
	public Tooltip() {}

	public String getLabel(){
		return label;
	}

	/**
	 * @deprecated In a future release, this getter will be removed since its value is not returned from the API.
	 */
	@Deprecated
	public String getImageURL(){
		return imageURL;
	}

	/**
	 * @deprecated In a future release, this getter will be removed since Tooltip no longer contains an image.
	 */
	@Deprecated
	public Drawable getImage(){
		return imageDrawable;
	}

	/**
	 * When passing a {@link PaymentItem} via an {@link android.content.Intent}, the tooltip image cannot be retrieved via this function.
	 * Instead you will need to retrieve the tooltip image in your own app using the URL returned from {@link #getImageURL()}.
	 *
	 * @return the tooltip image as a {@link Drawable}
	 *
	 * @deprecated In a future release, this getter will be removed since Tooltip no longer contains an image.
	 */
	@Deprecated
	public Drawable getImageDrawable() {
		return imageDrawable;
	}

	/**
	 * @deprecated In a future release, this getter will be removed since Tooltip no longer contains an image.
	 */
	@Deprecated
	public void setImageDrawable(Drawable imageDrawable){
		this.imageDrawable = imageDrawable;
	}
}

File: sync/Worldline/repo/onlinepayments-sdk/src/main/kotlin/com/onlinepayments/sdk/client/android/model/paymentproduct/Tooltip.kt
/*
 * Do not remove or alter the notices in this preamble.
 *
 * Copyright © 2025 Worldline and/or its affiliates.
 *
 * All rights reserved. License grant and user rights and obligations according to the applicable license agreement.
 *
 * Please contact Worldline for questions regarding license and user rights.
 */
package com.onlinepayments.sdk.client.android.model.paymentproduct

import java.io.Serializable

/**
 * Data class that represents a Tooltip object.
 * Tooltips are payment product specific and are used to show extra information about an input field.
 */
data class Tooltip(
    val label: String? = null
) : Serializable {

    companion object {
        @Suppress("Unused")
        private const val serialVersionUID = -317203058533669043L
    }
}
